packages/cms/lists/views/online-users.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useEffect, useState } from 'react'
+import axios from 'axios'
+import styled from 'styled-components'
+import { FieldProps } from '@keystone-6/core/types'
+import { FieldContainer } from '@keystone-ui/fields'
+import { controller } from '@keystone-6/core/fields/types/json/views'
+
+const colors = [
+  '#EEE8AA',
+  '#C0C0C0',
+  '#87CEEB',
+  '#F5DEB3',
+  '#90EE90',
+  '#FFB6C1',
+  '#FF8C00',
+  '#FFD700',
+  '#66CDAA',
+  '#FFE4B5',
+]
+
+const upateUserGql = `
+mutation Mutation($where: PostWhereUniqueInput!, $data: PostUpdateInput!) {
+  updatePost(where: $where, data: $data) {
+    onlineUsers {
+      email
+    }
+  }
+}
+`
+
+const currentUserGql = `
+query User {
+  authenticatedItem {
+    ... on User {
+      id
+      name
+      email
+    }
+  }
+}`
+
+const onlineUsersGql = `
+query($where: PostWhereUniqueInput!) {
+  post(where: $where) {
+    onlineUsers {
+      id
+      name
+      email
+    }
+  }
+}
+`
+
+const Container = styled(FieldContainer)`
+  display: flex;
+  flex-direction: row;
+`
+
+const Tooltip = styled.span`
+  visibility: hidden;
+  width: fit-content;
+  background-color: black;
+  color: #fff;
+  text-align: center;
+  border-radius: 6px;
+  padding: 5px 5px;
+  position: absolute;
+  z-index: 1;
+  bottom: -150%;
+  left: 0%;
+  margin-left: -60px;
+`
+
+const Avatar = styled.div`
+  width: 40px;
+  height: 40px;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border-radius: 20px;
+  margin: 0px 2px;
+  background: ${(props) => props.color};
+  position: relative;
+
+  &:hover ${Tooltip} {
+    visibility: visible;
+  }
+`
+
+type User = {
+  id: string
+  name: string
+  email: string
+}
+
+export const Field = ({ value }: FieldProps<typeof controller>) => {
+  let currentUserEmail = ''
```

Comment:
Suggestion:
Maybe `useRef` to replace js closure.

Reply:
fixed.